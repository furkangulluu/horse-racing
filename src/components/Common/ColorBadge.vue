<template>
  <div class="rounded-full w-4 h-4" :style="{ backgroundColor: color }"></div>
</template>

<script setup lang="ts">
const props = defineProps({
  color: {
    type: String,
    default: "",
  },
});
</script>
