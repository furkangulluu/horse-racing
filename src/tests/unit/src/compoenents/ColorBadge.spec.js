import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";
import ColorBadge from "@/components/Common/ColorBadge.vue";

describe("ColorBadge.vue", () => {
  test("applies the correct background color based on prop", () => {
    const color = "red";
    const wrapper = mount(ColorBadge, {
      props: {
        color,
      },
    });

    const div = wrapper.find("div");
    expect(div.attributes("style")).toContain(`background-color: ${color}`);
  });
});
