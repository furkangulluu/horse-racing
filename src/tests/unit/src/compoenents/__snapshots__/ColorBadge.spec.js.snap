// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseButton.vue > is correctly rendered 1`] = `"<div class="action-button flex justify-center items-center py-2 px-6 h-10 rounded-lg cursor-pointer ml-3 transition-all bg-off-white hover:bg-gray-200 text-black">Button</div>"`;
